pkg/scheduler/framework/plugins/noderesources/least_allocated.go

```diff
@@ -63,25 +64,40 @@ func (la *LeastAllocated) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewLeastAllocated initializes a new plugin and returns it.
-func NewLeastAllocated(_ runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+func NewLeastAllocated(laArgs runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+	args, ok := laArgs.(*config.NodeResourcesLeastAllocatedArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type NodeResourcesLeastAllocatedArgs, got %T", laArgs)
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	for _, resource := range (*args).Resources {
+		if resource.Weight <= 0 {
+			return nil, fmt.Errorf("resource Weight of %v should be a positive value, got %v", resource.Name, resource.Weight)
+		}
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
```

Comment:
should we place a maximum bound on the weight to avoid overflow errors when combining resources? for example, hardPodAffinitySymmetricWeight is limited to a max of 100

Reply:
oh true, and a limit of 100 was already mentioned in the godoc

Reply:
done